app/screens/thread/thread_follow_button/index.ts

```diff
@@ -6,17 +6,31 @@ import withObservables from '@nozbe/with-observables';
 import {of as of$} from 'rxjs';
 import {switchMap} from 'rxjs/operators';
 
-import {observeThreadById} from '@queries/servers/thread';
+import {observeTeamIdByThreadId, observeThreadById} from '@queries/servers/thread';
+import EphemeralStore from '@store/ephemeral_store';
 
 import ThreadFollowButton from './thread_follow_button';
 
 import type {WithDatabaseArgs} from '@typings/database/database';
 
-const enhanced = withObservables(['threadId'], ({threadId, database}: {threadId: string} & WithDatabaseArgs) => {
+type Props = WithDatabaseArgs & {
+    teamId?: string;
+    threadId?: string;
+};
+
+const enhanced = withObservables(['threadId'], ({teamId, threadId, database}: Props) => {
+    // Fallback in case teamId or threadId are not defined per navigation not setting the props bug.
+    const thId = threadId || EphemeralStore.getCurrentThreadId();
+    const tId = teamId ? of$(teamId) : observeTeamIdByThreadId(database, thId).pipe(
+        switchMap((t) => of$(t || '')),
+    );
```

Comment:
fix two: in case the props are undefined we get the correct values with what is stored in the ephemeral store.